### ISSUE TITLE
querySelectorAll() encountering problem with nested <li> tags

### ISSUE THREAD

**Issue by MarkKoenig95** [jsdom/jsdom#2675]
### Basic info:

- **Node.js version:10.16.0** 
- **jsdom version: 15.1.1** 

### Minimal reproduction case

```js

var i = 0;

function getInfo() {
  i = 0;

  var theirName;

  let summaryInfo = document.querySelectorAll(".listClass");
  Array.prototype.slice.call(summaryInfo).forEach(n => {
    i++;
    try {
      console.log('Started: ', i, ' n = ', n.innerHTML);

      theirName = n.querySelector(".nameClass").textContent;

      console.log(theirName);
    }
    catch (err) {
      console.error(err);
    }
  });
}

document.documentElement.innerHTML = `
<li class="listClass">
  <div>
    <li>
      <h3 class="nameClass">Name1</h3>
    </li>
  </div>
</li>

<li class="listClass">
  <div>
    <h3 class="nameClass">Name2</h3>
  </div>
</li>`;

getInfo();
```

### Output

```
console.log __tests__/index.test.js:40
    Started:  1  n =
      <div>
        </div>

  console.error __tests__/index.test.js:48
    TypeError: Cannot read property 'textContent' of null
        at Array.slice.call.forEach.n (C:\...)
        ...

  console.log __tests__/index.test.js:40
    Started:  2  n =
      <div>
        <h3 class="nameClass">Name2</h3>
      </div>


  console.log __tests__/index.test.js:44
    Name2
```

As you can see, the first 'li'  has another nested 'li'. As a result, jsdom is not reading what is actually inside the 'div' tag

With the second one however you can tell that it is accurately reading the inner contents of the 'li' with the class "listClass", including the contents of the 'div'

### How does similar code behave in browsers?

Simply put, I have no problem running the code in the browser. The whole point to this project was to set up a test suite to reference against offline  html pages. 

While the original html page is more complicated, if I had a page with this html I would get a result of:

Name1
Name2

I tried to make this as simple and clear as possible. If you have any questions or need any clarifications please let me know.

**Comment by domenic** [comment]
I get the same result in the console in browsers: https://jsbin.com/boyowubexa/edit?html,console,output

So this is not a bug in jsdom, as it is behaving the same as browsers. I suggest you debug your code so that it works in browsers and then try it in jsdom.
